packages/js/src/executors/node/node.impl.ts

```diff
@@ -73,9 +73,17 @@ export async function* nodeExecutor(
 
   // Re-map buildable workspace projects to their output directory.
   const mappings = calculateResolveMappings(context, options);
-
-  const outputFileName =
-    buildOptions.outputFileName ?? `${path.parse(buildOptions.main).name}.js`;
+  const matches = buildOptions.main.match(/^(.*)\/([^/]*)$/);
+  let outputFileName =
+    buildOptions.outputFileName || `${path.parse(buildOptions.main).name}.js`;
+
+  if (matches && !buildOptions.outputFileName) {
```

Comment:
It'd be better to move the match into the  `!buildOptions.outputFileName` if block, so we don't run the match unless required.